<?xml version="1.0" ?>
<launch>
    <arg name="leader_pose_topic" default='/virtual_leader/leader_pose'/>
    <arg name="leader_velocity_topic" default='/virtual_leader/leader_vel'/>
    <arg name="actual_pose_topic" default='/mir_pose_stamped_simple'/>
    <arg name="follower_cmd_vel_topic" default='/cmd_vel'/>
    <arg name="relative_position" default="[0.0, 1.0, 0.0]"/>

    <rosparam file="$(find formation_controller)/config/decentralized_leader_follower_control.yaml" command="load" subst_value="true"/>

	<node name="decentralized_leader_follower_control" pkg="formation_controller" type="decentralized_leader_follower_control.py" output="screen">
		<rosparam param="relative_position" subst_value="True">$(arg relative_position) </rosparam>
		<param name="leader_pose_topic" value="$(arg leader_pose_topic)"/>
        <param name="leader_velocity_topic" value="$(arg leader_velocity_topic)"/>
        <param name="actual_pose_topic" value="$(arg actual_pose_topic)"/>
        <param name="follower_cmd_vel_topic" value="$(arg follower_cmd_vel_topic)"/>
	</node>

</launch>